docs/proposals/source-verificiation-policies.md

```diff
@@ -0,0 +1,313 @@
+
+### Goals
+
+* Increase confidence in Git commit verification
+* Attach signer trust to sources, instead of at the app project level
+* Properly support multi-source applications as well as unique settings per source in multi-source applications
+* Be flexible enough to support other verification providers (e.g. Helm provenance, Git signed using sigstore, etc)
+* Make it easy for users to migrate to a more secure verification policy
+* Fully backwards compatibility to existing GnuPG commit verification mechanisms
+
+### Non-Goals
+
+* Implement other verification methods than GnuPG for Git commits. This may come in at a later stage, and the design of SVPs would allow those to integrate rather easily.
+
+## Proposal
+
+
+### Use cases
+
+Add a list of detailed use cases this enhancement intends to take care of.
+
+#### Use case 1:
+
+As an Argo CD user, I would like to ensure that my applications only syncs if every single commit in my source repository has been cryptographically signed by a trusted party.
+
+#### Use case 2:
+
+As an Argo CD user, I need to apply a different level of trust to different source repositories, especially with multi-source applications.
+
+## Implementation Details/Notes/Constraints
+
+There is a PoC implementation for this proposal, which is linked from the issue's description. This section describes the inner-workings of that PoC as envisioned by the authors.
+
+Under the hood, getting the revision history is performed using [git rev-list](https://git-scm.com/docs/git-rev-list), which can be consulted for more information.
+
+### Overview
+
+Source verification policies are configured in the `sourceVerificationPolicies` field of an AppProject's spec. The field itself is a list, and you can configure an arbitrary amount of source verification policies.
+
+Each entry in the list has the following mandatory fields:
+
+* `repositoryPattern` defines a glob-style pattern that an Application's source repository URL must be matched against in order for the policy to apply
+* `verificationLevel` is the verification level for the policy
+* `verificationMethod` is the verification method for the policy
+* `trustedSigners` is the list of allowed signers
+
+### Anatomy of a source verification policy
+
+A source verification policy describes requirements for a source, as used by an Argo CD Application, that have to be met in order to be used for a sync.
+
+A source verification policy has the following properties:
+
+* A repository pattern (shell-glob type) that is matched against the URL of the source (mandatory)
+* A repository type that is matched against the application's repository type (mandatory)
+* A verification level that describes with which level the verification should be performed (mandatory)
+* A verification methods that describes how to the source should be verified (mandatory)
+* A list of identities of trusted signers (optional, if omitted, any known signer is trusted)
```

Comment:
I feel like this list is a duplicate of the one above - could probably merge them.